### ISSUE TITLE
Add support for Ministral-8B-Instruct-2410

### ISSUE THREAD

**Issue by maralski** [EricLBuehler/mistral.rs#897]
https://huggingface.co/mistralai/Ministral-8B-Instruct-2410

```
(base) m@Peters-MacBook-Pro release % ./mistralrs-server -V    
mistralrs-server 0.3.2
(base) m@Peters-MacBook-Pro release % ./mistralrs-server -i plain -m mistralai/Ministral-8B-Instruct-2410
2024-11-05T04:00:20.056766Z  INFO mistralrs_server: avx: false, neon: true, simd128: false, f16c: false
2024-11-05T04:00:20.056793Z  INFO mistralrs_server: Sampling method: penalties -> temperature -> topk -> topp -> minp -> multinomial
2024-11-05T04:00:20.056799Z  INFO mistralrs_server: Model kind is: normal (no adapters)
2024-11-05T04:00:20.056947Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer.json` at `mistralai/Ministral-8B-Instruct-2410`
2024-11-05T04:00:20.057009Z  INFO mistralrs_core::pipeline::normal: Loading `config.json` at `mistralai/Ministral-8B-Instruct-2410`
2024-11-05T04:00:20.520147Z  INFO mistralrs_core::pipeline::paths: Found model weight filenames ["model-00001-of-00004.safetensors", "model-00002-of-00004.safetensors", "model-00003-of-00004.safetensors", "model-00004-of-00004.safetensors"]
2024-11-05T04:00:20.799849Z  INFO mistralrs_core::pipeline::normal: Loading `generation_config.json` at `mistralai/Ministral-8B-Instruct-2410`
2024-11-05T04:00:21.396891Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer_config.json` at `mistralai/Ministral-8B-Instruct-2410`
2024-11-05T04:00:21.398325Z  INFO mistralrs_core::pipeline::normal: Loading model `mistralai/Ministral-8B-Instruct-2410` on metal[4294968906].
2024-11-05T04:00:21.398448Z  INFO mistralrs_core::utils::log: Automatic loader type determined to be `mistral`
2024-11-05T04:00:21.406401Z  INFO mistralrs_core::utils::normal: DType selected is BF16.
2024-11-05T04:00:21.406468Z  INFO mistralrs_core::pipeline::normal: Model config: Config { vocab_size: 131072, hidden_size: 4096, intermediate_size: 12288, num_hidden_layers: 36, num_attention_heads: 32, num_key_value_heads: 8, hidden_act: Silu, max_position_embeddings: 32768, rms_norm_eps: 1e-5, rope_theta: 100000000.0, sliding_window: Some(32768), use_flash_attn: false, head_dim: Some(128), quantization_config: None, tie_word_embeddings: false }
100%|██████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 0.00it/s]100%|████████████████████████████████████████████████████████████████████████████| 117/117 [00:00<00:00, 137.85it/s]100%|████████████████████████████████████████████████████████████████████████████| 115/115 [00:00<00:00, 136.89it/s]100%|███████████████████████████████████████████████████████████████████████████████| 94/94 [00:00<00:00, 82.15it/s]
Error: data did not match any variant of untagged enum ModelWrapper at line 1217867 column 3
(base) m@Peters-MacBook-Pro release % 
```

**Comment by EricLBuehler** [comment]
@maralski I just merged #904, which now works for me. Can you please let me know if the following works?

```
git pull
cargo run --features metal --release '--' -i --isq Q4K plain -m mistralai/Ministral-8B-Instruct-2410
```

**Comment by maralski** [comment]
@EricLBuehler works great. Thanks

```
(base) m@Peters-MacBook-Pro mistral.rs % cargo run --features metal --release '--' -i --isq Q4K plain -m mistralai/Ministral-8B-Instruct-2410

    Finished `release` profile [optimized] target(s) in 0.69s
     Running `target/release/mistralrs-server -i --isq Q4K plain -m mistralai/Ministral-8B-Instruct-2410`
2024-11-10T06:35:40.541169Z  INFO mistralrs_server: avx: false, neon: true, simd128: false, f16c: false
2024-11-10T06:35:40.541193Z  INFO mistralrs_server: Sampling method: penalties -> temperature -> topk -> topp -> minp -> multinomial
2024-11-10T06:35:40.541241Z  INFO mistralrs_server: Model kind is: normal (no adapters)
2024-11-10T06:35:40.542271Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer.json` at `mistralai/Ministral-8B-Instruct-2410`
2024-11-10T06:35:40.542520Z  INFO mistralrs_core::pipeline::normal: Loading `config.json` at `mistralai/Ministral-8B-Instruct-2410`
2024-11-10T06:35:41.036430Z  INFO mistralrs_core::pipeline::paths: Found model weight filenames ["model-00001-of-00004.safetensors", "model-00002-of-00004.safetensors", "model-00003-of-00004.safetensors", "model-00004-of-00004.safetensors"]
2024-11-10T06:35:41.333350Z  INFO mistralrs_core::pipeline::normal: Loading `generation_config.json` at `mistralai/Ministral-8B-Instruct-2410`
2024-11-10T06:35:41.922789Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer_config.json` at `mistralai/Ministral-8B-Instruct-2410`
2024-11-10T06:35:41.925328Z  INFO mistralrs_core::pipeline::normal: Loading model `mistralai/Ministral-8B-Instruct-2410` on metal[4294968906].
2024-11-10T06:35:41.925730Z  INFO mistralrs_core::utils::log: Automatic loader type determined to be `mistral`
2024-11-10T06:35:41.934103Z  INFO mistralrs_core::utils::normal: DType selected is BF16.
2024-11-10T06:35:41.934195Z  INFO mistralrs_core::pipeline::normal: Model config: Config { vocab_size: 131072, hidden_size: 4096, intermediate_size: 12288, num_hidden_layers: 36, num_attention_heads: 32, num_key_value_heads: 8, hidden_act: Silu, max_position_embeddings: 32768, rms_norm_eps: 1e-5, rope_theta: 100000000.0, sliding_window: Some(32768), use_flash_attn: false, head_dim: Some(128), quantization_config: None, tie_word_embeddings: false }
100%|███████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00, 0.00it/s]100%|███████████████████████████████████████████████████████████████████████████████████████| 117/117 [00:07<00:00, 9.20it/s]100%|████████████████████████████████████████████████████████████████████████████████████████| 94/94 [00:07<00:00, 35.05it/s]100%|██████████████████████████████████████████████████████████████████████████████████████| 115/115 [00:07<00:00, 10.43it/s]
2024-11-10T06:35:49.697198Z  INFO mistralrs_core::pipeline::isq: Applying in-situ quantization into Some(Q4K) to 253 tensors.
2024-11-10T06:35:49.697232Z  INFO mistralrs_core::pipeline::isq: Applying ISQ on 1 threads.
2024-11-10T06:37:04.087168Z  INFO mistralrs_core::pipeline::isq: Applied in-situ quantization into Some(Q4K) to 253 tensors out of 253 total tensors. Took 74.39s
2024-11-10T06:37:04.218674Z  INFO mistralrs_core::pipeline::chat_template: bos_toks = "<s>", eos_toks = "</s>", unk_tok = <unk>
2024-11-10T06:37:04.222225Z  INFO mistralrs_server: Model loaded.
2024-11-10T06:37:04.222327Z  INFO mistralrs_server::interactive_mode: Starting interactive loop with sampling params: SamplingParams { temperature: Some(0.1), top_k: Some(32), top_p: Some(0.1), min_p: Some(0.05), top_n_logprobs: 0, frequency_penalty: Some(0.1), presence_penalty: Some(0.1), stop_toks: None, max_len: Some(4096), logits_bias: None, n_choices: 1, dry_params: Some(DrySamplingParams { sequence_breakers: ["\n", ":", "\"", "*"], multiplier: 0.0, base: 1.75, allowed_length: 2 }) }
====================
Welcome to interactive mode! Because this model is a text model, you can enter prompts and chat with the model.

Commands:
- `\help`: Display this message.
- `\exit`: Quit interactive mode.
- `\system <system message here>`:
    Add a system message to the chat without running the model.
    Ex: `\system Always respond as a pirate.`
====================
> How many r in strawberry
The word "strawberry" contains two "r" sounds. Here's the breakdown:

- "Straw" has one "r" sound.
- "Berry" has one "r" sound.

So, in total, there are two "r" sounds in the word "strawberry".
```
